trainers/base_trainer.py

```diff
@@ -109,20 +158,28 @@ def estimate_performance(self, eval_iters=None):
         for split in ["train", "val"]:
             losses = torch.zeros(eval_iters)
             perplexities = torch.zeros(eval_iters)
-            for i in range(eval_iters):
+            
+            ## init Pytorch's DataLoader
+            dataloader = self._prepare_dataloader(CustomDataset(self.dataloader.get_data(split), self.cfg.model["context_window"], self.cfg.general.device))
```

Comment:
prepare the dataloader once for train and eval and store these as attributes in the trainer. then just iterate over them. since the data doesn't change, there is no point of doing it like this

Reply:
which will be anyways what you do if you our dataloader inherit from the torch dataset